lightning/src/ln/chan_utils.rs

```diff
@@ -569,20 +569,23 @@ impl LocalCommitmentTransaction {
 		}
 	}
 
-	/// Add local signature for LocalCommitmentTransaction, do nothing if signature is already
-	/// present
+	/// Gets a local signature for the contained commitment transaction.
 	///
 	/// Funding key is your key included in the 2-2 funding_outpoint lock. Should be provided
 	/// by your ChannelKeys.
 	/// Funding redeemscript is script locking funding_outpoint. This is the mutlsig script
 	/// between your own funding key and your counterparty's. Currently, this is provided in
 	/// ChannelKeys::sign_local_commitment() calls directly.
 	/// Channel value is amount locked in funding_outpoint.
-	pub fn add_local_sig<T: secp256k1::Signing>(&mut self, funding_key: &SecretKey, funding_redeemscript: &Script, channel_value_satoshis: u64, secp_ctx: &Secp256k1<T>) {
-		if self.has_local_sig() { return; }
+	pub fn get_local_sig<T: secp256k1::Signing>(&self, funding_key: &SecretKey, funding_redeemscript: &Script, channel_value_satoshis: u64, secp_ctx: &Secp256k1<T>) -> Signature {
```

Comment:
This is used by any implementer of ChannelKeys, so I don't think so? I mean we should probably drop it wholesale because just writing the two lines in any implementer of ChannelKeys is better, but...